x/mongo/driver/topology/cmap_prose_test.go

```diff
@@ -76,20 +76,19 @@ func TestCMAPProse(t *testing.T) {
 				}
 				pool := createTestPool(t, cfg, connOpts...)
 
-				_, err := pool.get(context.Background())
-				assert.NotNil(t, err, "expected get() error, got nil")
+				_, err := pool.checkOut(context.Background())
+				assert.NotNil(t, err, "expected checkOut() error, got nil")
 
-				// If the connection doesn't finish connecting before resourcePool gives it back, the error will be
-				// detected by pool.get and result in a created/closed count of 1. If it does finish connecting, the
-				// error will be detected by resourcePool, which will return nil. Then, pool will try to create a new
-				// connection, which will also error. This process will result in a created/closed count of 2.
```

Comment:
Is this comment not still true of `pool.checkOut()`? I only ask because the two assertions below are still there.

Reply:
There is actually still a race condition between new connections requested by `maintain()` and new connections requested by `checkOut`, so the check needs to remain the same. I'll update the comment to reflect the new reasoning.

Reply:
Nice; the new tests look good to me here.